### DISCUSSION TITLE
GitHub Copilot doesn't use related files for suggestions in svelte files (JetBrains)

### DISCUSSION THREAD

**Discussion by NikolayMakhonin** [community/community#46246] [Copilot Conversations]
### Select Topic Area

Bug

### Body

# Description
I'm not sure if this is a bug, it could be a lack of functionality.
Github Copilot in [its documentation](https://docs.github.com/en/copilot/overview-of-github-copilot/about-github-copilot#about-github-copilot) states the following feature:
"GitHub Copilot analyzes the context in the file you are editing, as well as **related files**, and offers suggestions from within your text editor"

This feature (include related files to the context) works great in TypeScript files. But it doesn't work at all in `*.sevlte` files. It seems that Copilot treats `*.svelte` files in isolation from the whole project, as if I were writing code from scratch, although it must pull in the context from the imported file.

# Environment
* JetBrains IntelliJ IDEA 2022.1.2 (Ultimate Edition) Build #IU-221.5787.30, built on May 31, 2022
* GitHub Copilot plugin 1.2.1.2347
* Svelte plugin 0.22.1

# Reproducion

**TreeListNode.ts**
```
export interface ITreeListNode<Data = any> {
  // TODO
}

export function treeListNodeToObject<Data, TObject>(
    node: ITreeListNode<Data>,
    createObject: (data: Data, children: TObject[]) => TObject,
): TObject {
  // TODO
}

export function objectToTreeListNode<Data, TObject>(
    object: TObject,
    getChildren: (object: TObject) => TObject[],
    getData: (object: TObject) => Data,
): ITreeListNode<Data> {
  // TODO
}
```

**TreeListNode.test.ts**
```
import {ITreeListNode} from './TreeListNode'

describe('TreeListNode', function () {
    it('create with data', function () {
        let categories: ITreeListNode<any> = null
        let object = {
            name: 'root',
            children: [
                {
                    name: 'child1',
                },
            ],
        }

        categories = obj // The Copilot suggested correct code here: "objectToTreeListNode(object, (o) => o.children, (o) => o.name)"
    })
})
```

**Categories.svelte**
```
<script lang="ts">
    import {ITreeListNode, objectToTreeListNode} from './TreeListNode'

    let categories: ITreeListNode<any> = null
    let object = {
        name: 'root',
        children: [
            {
                name: 'child1',
            },
        ],
    }

    categories = obj // The Copilot suggested incorrect code here: "object"
    categories = objectToTreeListNode(object, o => // The Copilot suggested incorrect code here: "objectToTreeListNode(object, o => o.children, categories)"
</script>
```

**Comment by ZepposBE** [comment]
I have the same issue in .AL code.
Using a reference to another file/codeunit/module/... doesn't seem to influance copilot's access to the internal information (functions,...) of those files.
I've also created a post to have settings to be able to allow access to other files, access to dependencies of an app,  ... (No response so far)

**Comment by stylehouse** [comment]
Here's a smaller repro. I have this in VSCode so must be unrelated to JetBrains and perhaps also Svelte.

in a Svelte project, in src/lib/St.ts:

    export function detection(s) {
        let typ = {}
        if (s.such) typ.A = 1
        if (s.else) typ.B = 1
        return typ
    }

in src/routes/Code.svelte:

    import {detection}  from '$lib/St'
    let typ = detection({such:1})
    if (typ.

Then it seems to know nothing about typ.A|B
If you define detection() in Code.svelte it does suggest good.
Unrelated to the $lib alias (using ../lib/St also doesn't work).
It also seems unrelated to Svelte: I get bad completions even in Steez.ts, importing from './lib/St'.
I am new to the js world, my intuition from here is that vite (part of Svelte) may be complicating things (I know not how), but I would expect that the Svelte for VS Code (v107.3.0) plugin would know how to present that to Copilot.

What is limiting its view? https://github.com/orgs/community/discussions/13617 says "Just files adjacent to the current file in the ide." but that still doesn't do it (if I place St.ts and Steez.ts tabs next to each other?).

bonus bug: in VS Code 1.76.2, going to "Tweet Feedback" (at bottom-right of the window) -> "Submit a bug" -> Form to fill out -> "Preview on Github" leads to a 404: https://github.com/community/community/discussions/categories/copilot/issues/new?title=etcetc
seems like that is supposed to be:
https://github.com/orgs/community/discussions/new?category=copilot&title=etcetc
